testing/regress/ecl/mongodb-simple.ecl

```diff
@@ -0,0 +1,92 @@
+Import MongoDB;
+
+server := 'cluster0.uniqueserver.mongodb.net';
+
+username := 'hpccuser';
+
+password := 'supersecretpassword';
+
+analyticsrec := RECORD
+    INTEGER account_id;
+    INTEGER transaction_count;
+END;
+
+
+airbnbrec := RECORD
+    UTF8 name; 
+    UTF8 space; 
+    UTF8 description; 
+    INTEGER beds; 
+    INTEGER accommodates;
+END;
+
+feesrec := RECORD
+    DECIMAL32_2 price;
+    DECIMAL32_2 security_deposit;
+    DECIMAL32_2 cleaning_fee;
+    DECIMAL32_2 extra_people;
+    REAL guests_included;
+END;
+
+// The sample datasets "sample_analytics.transactions" and "sample_airbnb.listingsAndReviews" need to be loaded onto the cluster manually.
+// To load the sample datasets onto the cluster navigate to the "Database Deployments" page by clicking "Database" in the taskbar on the left.
+// Then click the button with three dots next to the cluster you wish to load them on. Then select "Load Sample Dataset" and wait for the data to be loaded.
+
+// Returns the document matching the account id that is passed in.
+dataset(analyticsrec) findTransactionsByID(INTEGER id) := Embed(mongodb : user(username), password(password), server(server), database('sample_analytics'),  collection('transactions'))
+    find_one({account_id: $id});
+ENDEMBED;
+
+// Returns the document with a transaction count greater than or equal to the number passed in.
+dataset(analyticsrec) findTransactionsByCount(INTEGER trans_count, INTEGER id) := EMBED(mongodb : user(username), password(password), server(server), database('sample_analytics'),  collection('transactions'))
+    find_one({transaction_count: {$gte: $trans_count}, account_id: $id});
+ENDEMBED;
+
+// Returns the document with a transaction count less than the number passed in.
+dataset(analyticsrec) findTransactionsBelowCount(INTEGER trans_count, INTEGER id) := EMBED(mongodb : user(username), password(password), server(server), database('sample_analytics'),  collection('transactions'))
+    find_one({transaction_count: {$lt: $trans_count}, account_id: $id});
+ENDEMBED;
+
+// Returns the document matching the id that is passed in.
+dataset(airbnbrec) findABByID(STRING id) := EMBED(mongodb : user(username), password(password), server(server), database('sample_airbnb'),  collection('listingsAndReviews'))
+    find_one({_id: $id});
+ENDEMBED;
+
+// Returns a dataset. Finds the document where the beds and accomodates fields are higher than the argument values. Matches _id to ensure the same document gets returned.
+dataset(airbnbrec) findABBeds(INTEGER beds, INTEGER accommodates, STRING id='10006546') := EMBED(mongodb : user(username), password(password), server(server), database('sample_airbnb'),  collection('listingsAndReviews'))
```

Comment:
Unfortunately, it cannot ignore the parameter if it is an empty string. It will create the document with the _id field having an empty string for its value which will match 0 zero documents. I could remove the _id parameter altogether, but then it will be possible for different documents to be returned.

Reply:
I could also change the commands to use different fields to get it down to a single document without having to use _id.

Reply:
can we use a different version of find_one({beds: {$gte: $beds}, accommodates: {$gte: $accommodates}); without "_id" if id is empty?
if not, it's not a big deal, force the caller to provide the proper id like you were doing before.

Reply:
I can do that, but I would have to create an additional findFeebyBeds function that has a different embed statement and use the ECL IF statement to decide which of the two functions to call. The plugin is kind of "dumb" and it only allows a single line in the embedded script and uses that assumption to build the documents. If that is favorable I can get going on that.

Reply:
fair enough, I'd force the user to provide the ID value

Reply:
Ok, I will make that change.